---
output: pdf_document
---

# **Analysis Resources**

<br>

The table below shows the description and version of Tools or files used in analysis at DNA Link.

<br>

Description                     |Information                      
--------------------------------|-------------------------------------
***Genom build***           	  | hg19/B37
***Aligner***                   | [Bowtie2[^1]](http://bowtie-bio.sourceforge.net/bowtie2/index.shtml) 
***Methylation Caller***        | [Bismark[^2]](https://www.bioinformatics.babraham.ac.uk/projects/bismark/)  
***DMR Analysis***              | [Biseq[^3]](https://bioconductor.org/packages/release/bioc/html/BiSeq.html)
***Annotation***                | [SnpEff[^4]](http://snpeff.sourceforge.net/)
***Target Region***             | [84Mb design covering 3.7 million CpGs](https://www.genomics.agilent.com/article.jsp?crumbAction=push&pageId=3038)

Table: Resources


[^1]: Langmead B, Salzberg S. Fast gapped-read alignment with Bowtie 2. Nature Methods. 2012, 9:357-359.
[^2]: Felix Krueger  Simon R. Andrews. Bismark: a flexible aligner and methylation caller for Bisulfite-Seq applications. 
Bioinformatics, Volume 27, Issue 11, 1 June 2011.
[^3]: Hebestreit K1, Dugas M, Klein HU. Detection of significantly differentially methylated regions in targeted bisulfite sequencing data. Bioinformatics. 2013 Jul 1;29(13):1647-53
[^4]: Cingolani P, Platts A, Wang le L, Coon M, Nguyen T, Wang L, Land SJ, Lu X, Ruden DM. "A program for annotating and predicting the effects of single nucleotide polymorphisms, SnpEff: SNPs in the genome of Drosophila melanogaster strain w1118; iso-2; iso-3.", Fly (Austin). 2012 Apr-Jun;6(2):80-92.


