#!/bin/sh
cutadapt -e 0.1 -q 51 -o $1.cutadapt			\
-b TACACTCTTTCCCTACACGACGCTCTTCCGATCT \
-b GAGCCGTAAGGACGACGACTTGGCGAGAAGGCTAG \
-b GATCGGAAGAGCGGTTCAGCAGGAATGCCGAG \
-b TCTAGCCTTCTCGCAGCACATCCCTTTCTCACAT \
-b AATGATACGGCGACCACCGAGATCTACACTCTTTCCCTACACGACGCTCTTCCGATCT \
-b TTACTATGCCGCTGGTGGCTCTAGATGTGAGAAAGGGATGTGCTGCGAGAAGGCTAGA \
-b TCTAGCCTTCTCGCCAAGTCGTCCTTACGGCTCTGGCTAGAGCATACGGCAGAAGACGAAC \
-b AGATCGGAAGAGCGGTTCAGCAGGAATGCCGAGACCGATCTCGTATGCCGTCTTCTGCTTG \
-b ACACTCTTTCCCTACACGACGCTCTTCCGATCT \
-b TCTAGCCTTCTCGCCAAGTCGTCCTTACGGCTCTGGC \
-a TGTGTGTGT \
-a GTGTGTGTG \
-a CACACACAC \
-a ACACACACA \
-a GAGAGAGAG \
-a CTCTCTCTC \
-a GGGGGGGGG \
-a AGAGAGAGA \
-a TCTCTCTCT \
-a GCGCGCGCG \
-a CGCGCGCGC \
-a CCCCCCCCC \
-O 8	 \
$1 >& $1.cutadapt.summary

# Adapter 1		
# Adapter 2		
# PCR Primer 1		
# PCR Primer 2 		
# Sequencing Primer 1	
# Sequencing Primer 2	

# $1 is fastq or fastq.gz
# --rest-file=REST_FILE
# --overlap=3

