---
params:
    dynamictitle: "Axiom Library Comparison"
    reportdate: !r Sys.Date()
output: 
        html_notebook:
#         html_document:
         toc: true
         toc_float: true
         depth: 3
#         number_section: true
#         theme: readable
         theme: united
         highlight: tango
#         fig_caption: true
#         df_print: paged
#         df_print: tibble
#         df_print: kable
#         fig_caption: yes 
#         fig_retina: 1 
#         code_download: false
         
#kable
title: "`r params$dynamictitle`"
#date: "`r params$reportdate`"
author: Park MinYoung(minmin@dnalink.com) in BI, DNA Link
    
---

***

```{r setting, echo=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo=FALSE, message=FALSE, warning=FALSE, fig.width=7)
options(DT.options = list(pageLength = 10, language = list(search = 'Filter:')))
#setwd("/home/adminrig/workspace.min/DNALink.PDX/oncoprint/201803.vcf/")
```

```{r library, include=FALSE}
library(pacman)
p_load(ggplot2, tidyr, dplyr, DT, scales, tibble, mosaic, ComplexHeatmap)
library(sunburstR)
library(parsetR)
library(knitr)
library(kableExtra)

options(knitr.table.format = "html") 


# http://rstudio.github.io/DT/extensions.html
```



```{r read_data_table, include=FALSE}
df<-read.table("out", header=T, sep="\t", stringsAsFactors = FALSE)
#?read.table

df$Region[df$Region == "aa-indel"]<-"frameshift"
df$Region[df$Region == "cds"]<-"exon"

df$Region[df$Region == "UTR-3"]<-"UTR"
df$Region[df$Region == "UTR-5"]<-"UTR"


str(df)
```


Target Library Path : `r getwd()`   

해당 library 와 비교 library 상의 overlap 및 annotation region 에 대한 비율을 확인하기 위한 리포트   

비교 library 와 variants overlap 유무   

 * Y 
 * N 

<br>

**Variant Count**  

<br>


```{r, results='asis'}
out <- xtabs( ~  DoExist + Region , data=df)

out %>% 
  t() %>% 
  as.data.frame.matrix() %>% 
  kable() %>% 
  kable_styling("striped", full_width = F, position="left") 
```


<br>

**parset**  

<br>


```{r}

parset(out, tension = 0.5, height = 400, width = 600)

```



<br>

**sunBurst**

<br>



```{r}

df_seq<-
  #mutate(table, sequence=paste(Is_AllSame, Is_Pass, Remark, Type, sep="-")) %>%
df %>% 
  mutate( sequence=paste(DoExist, Region, sep="-")) %>%
  group_by( sequence) %>% 
  dplyr::summarise( Count =n())
  
sunburst(df_seq)
```



